src/convert/traits.rs

```diff
@@ -127,3 +127,54 @@ impl<T: IntoWasmAbi> ReturnWasmAbi for T {
         self.into_abi()
     }
 }
+
+/// Enables blanket implementations of `WasmDescribe`, `IntoWasmAbi`,
+/// `FromWasmAbi` and `OptionIntoWasmAbi` functionality on boxed slices of
+/// types which can be converted to and from `JsValue` without conflicting
+/// implementations of those traits.
+///
+/// Implementing these traits directly with blanket implementations would
+/// be much more elegant, but unfortunately that's impossible because it
+/// conflicts with the implementations for `Box<[T]> where T: JsObject`.
+pub trait JsValueVector {
+    type ToAbi;
+    type FromAbi;
+
+    fn describe();
+    fn into_abi(self) -> Self::ToAbi;
+    fn none() -> Self::ToAbi;
+    unsafe fn from_abi(js: Self::FromAbi) -> Self;
+}
```

Comment:
You may be right. I don't really have the context anymore to properly defend my decisions. Maybe I did it this way simply because I thought it was neater. I think it's nice to have it bundled up into a trait like this, so that if anyone wants to implement the same functionality for another type, they know exactly what they have to do. But if you would rather have generic functions instead, I don't mind.